lib/workers/repository/update/pr/code-owners.spec.ts

```diff
@@ -22,13 +23,183 @@ describe('workers/repository/update/pr/code-owners', () => {
       expect(codeOwners).toEqual(['@jimmy']);
     });
 
+    it('respects orphan files', async () => {
+      fs.readLocalFile.mockResolvedValueOnce(
+        ['* @jimmy', 'yarn.lock'].join('\n')
+      );
+      git.getBranchFiles.mockResolvedValueOnce(['yarn.lock']);
+      const codeOwners = await codeOwnersForPr(pr);
+      expect(codeOwners).toEqual([]);
+    });
+
     it('returns more specific code owners', async () => {
       fs.readLocalFile.mockResolvedValueOnce(
         ['* @jimmy', 'package.json @john @maria'].join('\n')
       );
       git.getBranchFiles.mockResolvedValueOnce(['package.json']);
       const codeOwners = await codeOwnersForPr(pr);
-      expect(codeOwners).toEqual(['@john', '@maria']);
+      expect(codeOwners).toEqual(['@john', '@maria', '@jimmy']);
+    });
+
+    describe('returns more specific code owners in monorepos', () => {
+      const mockCodeOwners = codeBlock`
+        # By default, assign to @john
+        #
+        * @john
+
+        # Lockfiles are not owned by anyone, any package dependency update may modify them.
+        # Assigning lockfiles an owner will cause issues as merge requests to be assigned to incorrect users
+        yarn.lock
+
+        # Assign each package to it's respective user
+        #
+        packages/a/ @maria
+        packages/b/ @jimmy
+        packages/c/ @dan
+        packages/d/ @maria @jimmy
+        packages/e/ @jimmy
```

Comment:
This may just be a "me problem": it's hard to keep track of _who_ owns which file(s).

Can we use more descriptive names for the `packages/*` or the `@users`? This could help make things easier to remember maybe?

Reply:
um yeah, I just wanted to keep the usernames from the tests that were already there, but we could change the package names and or usernames to have more descriptive names. Any suggestions?

Reply:
I'm sorry, this is too complicated for me to figure out. I hope one of the maintainers can help!